packages/flutter/lib/src/foundation/tree_diagnostics_mixin.dart

```diff
@@ -40,8 +1469,47 @@ String describeEnum(Object enumEntry) {
   return description.substring(indexOfDot + 1);
 }
 
-/// A mixin that helps dump string representations of trees.
-abstract class TreeDiagnosticsMixin {
+/// Returns a lower case hyphen separated version of a camel case name.
+///
+/// For example:
+///
+/// ```dart
+///
+/// main() {
+///   assert(toHyphenedName('deferToChild') == 'defer-to-child');
+///   assert(toHyphenedName('Monday') == 'monday');
+///   assert(toHyphenedName('monday') == 'monday');
+/// }
+/// ```
+String toHyphenedName(String word) {
+  final String lowerWord = word.toLowerCase();
+  if (word == lowerWord)
+    return word;
+
+  final StringBuffer sb = new StringBuffer();
```

Comment:
FWIW we usually use "buffer" for string buffers or, if it's what we're about to return, "result".